src/clic-formatter.ads

```diff
@@ -0,0 +1,22 @@
+package CLIC.Formatter
+with Preelaborate
+is
+
+   -- Provides markdown or TTY formatting for displaying help pages.
+
+   procedure Force_Disable_TTY;
+
+   procedure Enable_Markdown;
```

Comment:
For me this is not the right way to add a Markdown output for the help pages. You may want to print the help in markdown without changing the formatting of the other outputs of your program.

Also it  doesn't make much sens to outputting markdown for other parts of a command line program.

What I did for my [Man page POChttps://github.com/alire-project/clic/pull/12/files)) is to add a special `Print_Man_Page` to the `CLIC.Subcommand.Instance` package.

Reply:
But I like the idea of a Markdown output for the help :)

Reply:
My rationale for doing it through the formatting parameters of `CLIC.Subcommand.Instance` is that all the logic already present in `Display_Help` can be reused. In fact, when looking at that generic parameters, I figured out that something like I'm doing was envisioned.

For example, the recent change about the global switches (a6bea52a98602fb93dadee8332738f5ff6207a16) would have to be applied in two places if there is a different help page generation for both formats like in #12. Isn't it better if there is a single place where the page is composed and the format can be parameterized?

> You may want to print the help in markdown without changing the formatting of the other outputs of your program.
But the help printing is a single purpose command. When you print the help page, either for generating a markdown format or for directly reading, this will be the single action performed by the application, it's doesn't make sense to generate any other output.

Where I agree undoubtedly is that, instead of a global switch (`--markdown`) to be combined with `--help` as I thought it, it should be a single switch for a specific subcommand. I was influence by how `--no-tty` and similar switches worked and implemented it similarly, but always thinking in this `--markdown` switch to be used along with `--help` or the `help` subcommand, not for any other program output.

Reply:
> Isn't it better if there is a single place where the page is composed and the format can be parameterized?

This solution does work with TTY and Markdown, but it's not going to work for Man or HTML because of more complex output structures (lists in HTML for instance). So we will need another ad-hoc way to have those formats.

@mosteo up to you.

Reply:
If Markdown works well, HTML is indeed the final output of the former.

But, in any case, if @mosteo thinks the same, I don't have any problem in changing it to the other approach.

Reply:
I think we should emit a single format that is suitable for conversion with e.g. `pandoc` rather than having multiple generators. If the logic can be reused for the single chosen format, I'd prefer that, but if as @Fabien-Chouteau says, there are structures that won't work, then that forces our hand to have a separate generator. 

The way I'd go about it would be to use a sample complex page to check both issues: that it can be converted properly and whether the existent logic can be reused.